src/TMap.cpp

```diff
@@ -2380,6 +2398,29 @@ void TMap::slot_replyFinished(QNetworkReply* reply)
                 file.flush();
                 file.close();
 
+                if (file.fileName().endsWith(QStringLiteral("xml"), Qt::CaseInsensitive)) {
+                    auto pHost = mpHost;
+                    if (!pHost) {
+                        cleanup();
+                        return;
+                    }
+
+                    QString infoMsg = tr("[ INFO ]  - ... map downloaded and stored, now parsing it...");
+                    postMessage(infoMsg);
+                    if (pHost->mpConsole->loadMap(file.fileName())) {
```

Comment:
:beetle: This is **wrong** - `TConsole::loadMap(...)` is for binary format files - and you have already determined it is an XML one - and those are already processed in the code immediately afterwards - so it is not clear what you are trying to do here at all.

Reply:
Ha! Thanks for spotting it.